### PR TITLE
[11.x] Fix doc block and  refactor `Check` and `Algorithm` methods

### PR DESCRIPTION
**Pull request by saMahmoudzadeh**
#### Refactor `Check` method:

The original condition `strlen($hashedValue) === 0` checks whether the length of `$hashedValue` is equal to 0, indicating an empty string. I used `$hashedValue === ''` which is faster and more readable than `strlen($hashedValue) === 0` . So:

```php

       // before
       if (is_null($hashedValue) || strlen($hashedValue) === 0) {
            return false;
        }

       // after
       if (is_null($hashedValue) || $hashedValue === '') {
            return false;
        }
```

#### Refactor `algorithm` method:
In this method, there is also a mistake in the doc block where it states `@return int`, which should be `@return string`. Additionally, since this method returns a constant value which inherently and explicitly is a string,
the return type of the function can be set to **string**

#### After
```php 
   /**
     * Get the algorithm that should be used for hashing.
     *
     * @return int
     */
    protected function algorithm()
    {
        return PASSWORD_ARGON2ID;
    }
```
#### Before

```php 
   /**
     * Get the algorithm that should be used for hashing.
     *
     * @return string
     */
    protected function algorithm(): string
    {
        return PASSWORD_ARGON2ID;
    }
```

### Files Changed

- `src\illuminate\Hashing\Argon2IdHasher.php`
- `src\illuminate\Hashing\ArgonHasher.php`

### CODE REVIEW

**Comment by driesvints** [comment]
These aren't strings? https://3v4l.org/O6hp0

**Comment by saMahmoudzadeh** [comment]
@driesvints 

https://www.php.net/manual/en/password.constants.php#constant.password-argon2i

these are string

image

**Comment by saMahmoudzadeh** [comment]
@driesvints 

image

File: src/Illuminate/Hashing/ArgonHasher.php [hunk 1]
```diff
@@ -76,9 +76,9 @@ public function make($value, array $options = [])
     /**
      * Get the algorithm that should be used for hashing.
      *
-     * @return int
+     * @return string
      */
-    protected function algorithm()
+    protected function algorithm(): string
```
[the remark this answers is not in the document]
Comment (saMahmoudzadeh) [reply src/Illuminate/Hashing/ArgonHasher.php:81]: > Changing the return type on a public method is a breaking change. You should just stick to docblock changes.

In cases where the output is explicitly defined and there is no impact from any input, such as returning a simple constant, we can strictly type them. Apart from these cases, I fully agree with you.


But overall, it seems to me that as we move towards the future, it's better to lean towards strict typing.

**Comment by driesvints** [comment]
Re-attempt this without the return type signature changes, thanks.

**Pull request closed without merging**